rtv/history.py

```diff
@@ -0,0 +1,37 @@
+import os
+
+
+__all__ = ['load_history', 'save_history']
+
+
+def history_path():
+    """
+    Create the path to the history log
+    """
+    HOME = os.path.expanduser('~')
+    XDG_CONFIG_HOME = os.getenv('XDG_CACHE_HOME', os.path.join(HOME, '.config'))
+    path = os.path.join(XDG_CONFIG_HOME, 'rtv', 'history.log')
+    return path
+
+
+def load_history():
+    """
+    Load the history file into memory if it exists
+    """
+    path = history_path()
+    if os.path.exists(path):
+        with open(path) as history_file:
+            return set([line.replace('\n', '') for line in history_file])
+    return set()
+
+
+def save_history(history):
+    """
+    Save the visited links to the history log
+    """
+    path = history_path()
+    with open(path, 'w+') as history_file:
+        for i in range(200):
+            if not history:
+                break
+            history_file.write(history.pop() + '\n')
```

Comment:
Using `pop()` on a set returns an arbitrary element, and does not guarantee order. I'm pretty sure you're going to have to use a list for that. One way to solve this would be to keep two data structures internally, one to store the order and one to test membership. Here's something that comes to mind. Not saying you have to do this, it's just the first idea I had.

```python
class History(object):

    def __init__(self, ...):
        self._hlist = []
        self._hset = {}

    def __contains__(self, item):
        return (item in self._hset)

    def add(self, item):
        self._hlist.append(item)
        self._hset.add(item)

    def load(...)
    def save(...)
    def path(...)
```

Another nice side-effect of switching to a class is that you could take your 
```
@atexit.register
save()
``` 
call and stick it right inside of the `load()` method.